pkg/template/servicebroker/servicebroker.go

```diff
@@ -25,85 +25,112 @@ import (
 // Broker represents the template service broker.  It implements
 // openservicebroker/api.Broker.
 type Broker struct {
+	// privilegedKubeClientConfig is a privileged clientconfig used for communicating back to the
+	// API server.  This is needed because the template service broker is a fake server component that actually
+	// needs both the apiserver and controllers to be running
+	privilegedKubeClientConfig restclient.Config
+	initialize                 sync.Once
+
 	kc                 kclientset.Interface
 	templateclient     internalversiontemplate.TemplateInterface
 	extkc              kclientsetexternal.Interface
 	extrouteclient     extrouteclientset.RouteV1Interface
 	lister             templatelister.TemplateLister
+	hasSynced          func() bool
 	templateNamespaces map[string]struct{}
 	ready              chan struct{}
 }
 
 var _ api.Broker = &Broker{}
 
-// NewBroker returns a new instance of the template service broker.  While
+// DeprecatedNewBrokerInsideAPIServer returns a new instance of the template service broker.  While
 // built into origin, its initialisation is asynchronous.  This is because it is
 // part of the API server, but requires the API server to be up to get its
 // service account token.
-func NewBroker(privrestconfig restclient.Config, privkc kclientset.Interface, infraNamespace string, informer templateinformer.TemplateInformer, namespaces []string) *Broker {
+func DeprecatedNewBrokerInsideAPIServer(privilegedKubeClientConfig restclient.Config, informer templateinformer.TemplateInformer, namespaces []string) *Broker {
 	templateNamespaces := map[string]struct{}{}
 	for _, namespace := range namespaces {
 		templateNamespaces[namespace] = struct{}{}
 	}
 
 	b := &Broker{
+		privilegedKubeClientConfig: privilegedKubeClientConfig,
 		lister:             informer.Lister(),
+		hasSynced:          informer.Informer().HasSynced,
 		templateNamespaces: templateNamespaces,
 		ready:              make(chan struct{}),
 	}
 
-	go func() {
-		// the broker is initialised asynchronously because fetching the service
-		// account token requires the main API server to be running.
+	return b
+}
+
+// MakeReady actually makes the broker functional
+func (b *Broker) MakeReady() error {
+	select {
+	case <-b.ready:
```

Comment:
> I believe this is wrong and that b.ready should be removed.

You can't do that yet since the embedded templateservicebroker (still present) relies on it.  In the end, yes, it will be gone, but I was trying to spare us a broker intermediate step.